Palace/Utilities/System/LocationManager.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  LocationManager.swift
+//  Palace
+//
+//  Created by Maurice Carrier on 6/6/23.
+//  Copyright © 2023 The Palace Project. All rights reserved.
+//
+
+import CoreLocation
+
+extension Notification.Name {
+  static let locationAuthorizationDidChange = Notification.Name("LocationAuthorizationDidChange")
+}
+
+class LocationManager: NSObject, CLLocationManagerDelegate {
+  static let shared = LocationManager()
+  private let locationManager = CLLocationManager()
+
+  private override init() {
+    super.init()
+    locationManager.delegate = self
+  }
+  
+  var locationAccessAuthorized: Bool {
+    let status = locationManager.authorizationStatus
+    return status == .authorizedAlways || status == .authorizedWhenInUse
```

Comment:
When `authorizationStatus` is `.notDetermined`, the app shows "Open Settings" message when the app should ask the permission instead: on the first run (in this case Location access doesn't exist in the settings, so the user can't enable it) and when the user changes the value in the settings to "Ask next time or when I share".

Reply:
nice catch, updated